OneCommunityGlobal/HighestGoodNetworkApp
# Yihan send review email

# Description
<img width="735" alt="截屏2023-07-25 19 15 37" src="https://github.com/OneCommunityGlobal/HighestGoodNetworkApp/assets/94303659/3cb34b12-cfbc-40e6-9c90-224e35ee368b">


## Related PRS (if any):
This frontend PR is related to the [#482 ](https://github.com/OneCommunityGlobal/HGNRest/pull/482)backend PR.
This PR is based on [PR#1013](https://github.com/OneCommunityGlobal/HighestGoodNetworkApp/pull/1013). I recommand to test #1013 before review this one.

## Main changes explained:
- Update file ReviewButton.jsx for executing send email function after click on "submit" button.

## How to test:

Before moving to testing, firstly you need to set up your own test email id to send the mail notification.
This is only for test purposes. Once tested you can revert these changes. These changes are not required on the main.

To send email using Gmail API using OAuth2 and nodemailer, kindly follow the below video and setup up a test mail id.
https://www.youtube.com/watch?v=-rcRf7yswfM&t=10s&ab_channel=MafiaCodes

Fill below variable values from the above setup in the .env file of the backend.
REACT_APP_EMAIL =
REACT_APP_EMAIL_CLIENT_ID =
REACT_APP_EMAIL_CLIENT_SECRET =
REACT_APP_EMAIL_REFRESH_TOKEN =
REACT_APP_EMAIL_CLIENT_REDIRECT_URI = https://developers.google.com/oauthplayground
sendEmail = TRUE

(resource: #964 )

============================================================================================

1. check into current branch
2. do `npm install` and `...` to run this PR locally
3. log as any user
4. go to dashboard→ Tasks → "Submit for review" → "Submit"
5. verify you, other Amin/Mentor/Manager users on the same team and highestgoodnetwork@gmail.com receive the review request email.

## Screenshots or videos of changes:

Submit test 2 for review: 
<img width="863" alt="截屏2023-07-26 15 34 40" src="https://github.com/OneCommunityGlobal/HighestGoodNetworkApp/assets/94303659/87e41bb1-b9ec-4630-bd69-3f7a16982a70">

Then receive the email:
<img width="1060" alt="截屏2023-07-26 15 30 28" src="https://github.com/OneCommunityGlobal/HighestGoodNetworkApp/assets/94303659/7ca0161e-78e0-40df-8639-7cd1eef58670">
